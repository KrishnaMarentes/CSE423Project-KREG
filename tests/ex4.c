int main() {
    int j=4;
    float k = 2;
    f = 2 == 2;
    printf("hello, world!", f);

    return calc()+2;
}