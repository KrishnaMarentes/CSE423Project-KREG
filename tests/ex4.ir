function calc()
{
int f;

f = 0;
return f;
}

function main()
{
int j;
float k;

j = 4;
k = 2;
KREG.0 = calc();
KREG.1 = KREG.0 + 2;
return KREG.1;
}



