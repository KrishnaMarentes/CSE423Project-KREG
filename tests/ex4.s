.section .text
	.global _start

calc:
	movl %esp, %ebp
	popl %ebp
	ret
_start: 
	call calc
