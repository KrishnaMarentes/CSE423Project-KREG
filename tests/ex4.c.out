program 
  funDeclaration 
    typeSpecifier int
    main
    varDeclaration 
      typeSpecifier int
      varDeclList 
        varDeclInitialize 
          varDeclId j
          =
          constant 4
    varDeclaration 
      typeSpecifier float
      varDeclList 
        varDeclInitialize 
          varDeclId k
          =
          constant 2
    expression 
      mutable f
      =
      relExpression 
        constant 2
        relop ==
        constant 2
    call printf
      args 
        argList 
          argList 
            constant "hello, world!"
          ,
          mutable f
    returnStmt return
      sumExpression 
        call calcargs
        sumop +
        constant 2
